### PR TITLE
adding the initial set of files

### PR DESCRIPTION
**Pull request by joyceg**
Adding the initial set of values to get reviewed.

### CODE REVIEW

File: config/install/search_config.settings.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+containing_any: ''
+containing_phrase: ''
+containing_none: ''
+types: ''
+category: ''
+language: ''
+limit: 10
+admin_bypass: 1
+remove_advanced: 0
+search_config_string_overrides: []
```
Comment (naveenvalecha) [inline config/install/search_config.settings.yml]: add a new line at the bottom

File: config/schema/search_config.settings.schema.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
+search_config.settings:
+	type:config_object
```
Comment (naveenvalecha) [inline config/schema/search_config.settings.schema.yml:2]: convert tabs to spaces in whole file.

File: search_config.info.yml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+name: Search Configuration
+description: Modify search form, including enforcing search restrictions by content type.
```
Comment (naveenvalecha) [inline search_config.info.yml:2]: put the description value in single/double quotes

File: search_config.install [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+/*
+ *Implements hook_schema
+ */
+
+function search_config_schema() {
+  $schema['search_config_exclude'] = array(
+    'description' => 'Provides a way to exclude specific entities from the search results. Note that this does not stop the entities from being indexed.',
+    'fields' => array(
+      'entity_id' => array(
+        'description' => 'The {node}.nid to exclude.',
```
Comment (naveenvalecha) [inline search_config.install:12]: the description should not refer node schema b/c entity id can vary for different content entities.

File: search_config.install [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+          $insert->values(array(
+            'rid' => $rid,
+            'permission' => 'search all content',
+          ));
+        }
+      }
+      $insert->execute();
+      $settings = search_config_node_settings();
+      \Drupal::service('config.factory')->getEditable('search_config.settings')->save();
+    }
```
Comment (naveenvalecha) [inline search_config.install:58]: Add a new line.

File: search_config.module [hunk 6]
```diff
@@ -0,0 +1,166 @@
+<?php
+
+/*
+ * Helper function for the installation of the module
```
Comment (naveenvalecha) [inline search_config.module:4]: Remove the old legacy code from .module file which we are not using

File: templates/search-config-admin-labels-form.tpl.php [hunk 7]
```diff
@@ -0,0 +1,30 @@
+<?php
```
Comment (naveenvalecha) [inline templates/search-config-admin-labels-form.tpl.php:1]: Why this .tpl.php file ?

Comment (joyceg) [inline templates/search-config-admin-labels-form.tpl.php:1]: I need to replace it with the twig.

Comment (naveenvalecha) [inline templates/search-config-admin-labels-form.tpl.php:1]: remove it at first . As I said you to keep the repo clean

**Comment by joyceg** [comment]
Closing this pull request.
Added a second one on branch 8.x-1.x

**Comment by neetumorwani** [comment]
This PR is obsolete and is  not needed. @joyceg make sure you are working on 8.x.-1.x throughput the project

**Pull request closed without merging**